packages/js/product-editor/src/components/number-control/number-control.tsx

```diff
@@ -72,7 +77,51 @@ export const NumberControl: React.FC< NumberProps > = ( {
 				step={ step }
 				disabled={ disabled }
 				id={ id }
-				suffix={ suffix }
+				suffix={
+					<>
+						{ suffix }
+						{ showStepButtons && (
+							<>
+								<Button
+									icon={ plus }
+									onClick={ () =>
+										onChange(
+											String(
+												parseFloat( value || '0' ) +
+													step
+											)
+										)
+									}
+									isSmall
+									aria-hidden="true"
+									aria-label={ __(
+										'Increment',
+										'woocommerce'
+									) }
+									tabIndex={ -1 }
+								/>
+								<Button
+									icon={ reset }
+									onClick={ () =>
+										onChange(
+											String(
+												parseFloat( value || '0' ) -
+													step
+											)
+										)
+									}
+									isSmall
+									aria-hidden="true"
+									aria-label={ __(
+										'Decrement',
+										'woocommerce'
+									) }
+									tabIndex={ -1 }
+								/>
+							</>
+						) }
+					</>
+				}
```

Comment:
I wonder if we should be doing this, or whether we should:

- Set `type="number"` on the `InputControl`
- Leave the handling of stepper buttons to the core `InputControl` component (which currently just uses the browser default).

With this implementation, we are not letting the browser (1) know that we are expecting only numeric input, and (2) control the stepper display.

I'm concerned that WP core will change stepper handling in the future, which we wouldn't get for free. And, we are already diverging from how steppers look elsewhere in the WP/WC experience, since they are using the built-in steppers.

(Note, I kinda like our stepper design, but it does have the confusion of "+" also meaning "add a new item", and "-" also meaning "remove an item".)

Reply:
If we keep our custom steppers, I do think we should modify the implementation to only show them when the field is focused/hovered on (like the native steppers do). This will reduce visual noise when viewing the form but not editing the specific field.